File: trunk/openrest4j-api/src/com/googlecode/openrest/Item.java
package com.googlecode.openrest;

import java.io.Serializable;
import java.util.Collections;
import java.util.List;

import org.codehaus.jackson.annotate.JsonIgnoreProperties;
import org.codehaus.jackson.map.annotate.JsonSerialize;

/**
 * An item that can be ordered, e.g. a main dish ("hamburger"), a side ("fries")
 * or a dish variation ("well done").
 * @author DL
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class Item implements Serializable {
    public Item(String id, String restaurantId, String title, String description,
            Integer price, List<com.googlecode.openrest.Variation> variations,
            com.googlecode.openrest.Availability availability, Boolean inactive) {
        this.id = id;
        this.restaurantId = restaurantId;
        this.title = title;
        this.description = description;
        this.price = price;
        this.variations = variations;
        this.availability = availability;
        this.inactive = inactive;
    }

    /** Default constructor for JSON deserialization. */
    public Item() {}

    /** The item's unique id. */
    @JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
    public String id;

    /** The restaurant's id. */
    @JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
    public String restaurantId;

    /** The item's name. */
    @JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
    public String title;

    /** The item's one line description. */
    @JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
    public String description;

    /** The item's price, in "cents". */
    @JsonSerialize(include = JsonSerialize.Inclusion.NON_DEFAULT)
    public Integer price = 0;

    /** List of possible variations. */
    @JsonSerialize(include = JsonSerialize.Inclusion.NON_DEFAULT)
    public List<Variation> variations = Collections.emptyList();

    /** Time windows in which this item is regularly available. */
    @JsonSerialize(include = JsonSerialize.Inclusion.NON_DEFAULT)
    public Availability availability = new Availability();

    /** Human readable availability string. */
    @JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
    public String availabilityStr;

    /** Whether the item is deactivated (i.e. suspended or disabled). */
    @JsonSerialize(include = JsonSerialize.Inclusion.NON_DEFAULT)
    public Boolean inactive = Boolean.FALSE;

    /** The current status. */
    @JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
    public Status status;
    
    private static final long serialVersionUID = 1L;
}

File: openrest4net/openrest4net-api/com/openrest/v1_1/Item.cs
using System;
using System.Collections.Generic;
using com.openrest.v1_1;

namespace com.openrest.v1_1
{
    /**
     * An item that can be ordered, e.g. a main dish ("hamburger"), a side ("fries")
    * or a dish variation ("well done").
    * @author DL
    */
    public class Item
    {
        public Item(string id, string restaurantId, string title, string description,
            int price, IList<Variation> variations, Availability availability, bool inactive, string picture)
        {
            this.id = id;
            this.restaurantId = restaurantId;
            this.title = title;
            this.description = description;
            this.variations = variations;
            this.availability = availability;
            this.inactive = inactive;
            this.picture = picture;
        }

        /** Empty constructor required for initialization from JSON-encoded string. */
        public Item() { }

        /** The item's unique id. */
        public string id;

        /** The restaurant's id. */
        public string restaurantId;

        /** The item's name. */
        public string title;

        /** The item's one line description. */
        public string description;

        /** The item's price, in "cents". */
        public int? price = 0;

        /** List of possible variations. */
        public IList<Variation> variations = new List<Variation>();

        /** Time windows in which this item is regularly available. */
        public Availability availability = new Availability();

        /** Human readable availability string. */
        public string availabilityStr;

        /** Whether the item is deactivated (i.e. suspended or disabled). */
        public bool? inactive = false;
        
        /**
         * Map of externally-defined item ids referring to this item.
         * For example, the item-id in the restaurant's PoS system.
         * 
         * Developers should use unique keys, e.g. "com.company.product".
         */
        public IDictionary<string, string> externalIds = new Dictionary<string, string>();

        /** The current status. */
        public Status status;

        /** Item picture URL (direct link). */
        public string picture;
    }
}
